 Ensure optarg is incremented strictly before it is read from.
 https://skarnet.org/lists/supervision/2971.html

--- runit-2.1.2/src/chpst.c.orig
+++ runit-2.1.2/src/chpst.c
@@ -308,6 +308,7 @@
     case 'n':
       switch (*optarg) {
         case '-':
+          ++optarg;
           if (optarg[scan_ulong(++optarg, &ul)]) usage(); nicelvl =ul;
           nicelvl *=-1;
           break;
